machine/machine_core/machine_virtual.py

```diff
@@ -543,34 +549,34 @@ def add_disk(self, size=None, serial=None, path=None, type='raw'):
         if self._domain.attachDeviceFlags(disk_desc, libvirt.VIR_DOMAIN_AFFECT_LIVE) != 0:
             raise Failure("Unable to add disk to vm")
 
-        disk = {
-            "path": image,
-            "serial": serial,
-            "filename": image,
-            "dev": dev,
-            "index": index,
-            "type": type,
-        }
+        disk = Disk(
```

Comment:
This value is returned, thus this is an API break. Please run some cockpit and c-machines tests against this to make sure this works. Otherwise, let's maybe dial this back.

Reply:
https://github.com/cockpit-project/cockpit/pull/17801 Cockpit test